File: engine/runtime-parser/src/main/scala/org/enso/compiler/core/ir/module/Scope.scala
package org.enso.compiler.core.ir.module

import org.enso.compiler.core.IR
import org.enso.compiler.core.ir.{Expression, IdentifiedLocation}

/** A representation of constructs that can only occur in the top-level
  * module scope
  */
trait Scope extends IR {

  /** @inheritdoc */
  override def mapExpressions(
    fn: java.util.function.Function[Expression, Expression]
  ): Scope

  /** @inheritdoc */
  override def setLocation(location: Option[IdentifiedLocation]): Scope

  /** @inheritdoc */
  override def duplicate(
    keepLocations: Boolean   = true,
    keepMetadata: Boolean    = true,
    keepDiagnostics: Boolean = true,
    keepIdentifiers: Boolean = false
  ): Scope
}

File: engine/runtime-parser/src/main/java/org/enso/compiler/core/ir/module/Scope.java
package org.enso.compiler.core.ir.module;

import java.util.function.Function;
import org.enso.compiler.core.IR;
import org.enso.compiler.core.ir.Expression;
import org.enso.compiler.core.ir.IdentifiedLocation;
import scala.Option;

/** A representation of constructs that can only occur in the top-level module scope */
public interface Scope extends IR {

  @Override
  Scope mapExpressions(Function<Expression, Expression> fn);

  @Override
  Scope setLocation(Option<IdentifiedLocation> location);

  @Override
  Scope duplicate(
      boolean keepLocations,
      boolean keepMetadata,
      boolean keepDiagnostics,
      boolean keepIdentifiers);
}
